src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -509,6 +511,19 @@ class ApplicationManagerImpl
    */
   void OnPTUFinished(const bool ptu_result) FINAL;
 
+  void ProcessFailedPTU() OVERRIDE;
+
+  void ProcessFailedStatusUpdate(
+      hmi_apis::Common_ServiceType::eType service_type,
+      hmi_apis::Common_ServiceEvent::eType service_event,
+      hmi_apis::Common_ServiceUpdateReason::eType service_update_reason) FINAL;
+
+  void ProcessSuccessfulStatusUpdate(
+      hmi_apis::Common_ServiceType::eType service_type,
+      hmi_apis::Common_ServiceEvent::eType service_event) FINAL;
+
+  void OnPTUFailed() FINAL {}
```

Comment:
@LuxoftAKutsan  `ProcessFailedPTU` is an interface that allows notifying `protocol_handler` that PTU was unsuccessful. On the other hand, `OnPTUFailed` is a part of `SecurityManagerListener` listener interface. This callback exists, so that `HandshakeHandler`  (which implements `SecurityManagerListener`) can be notified of failed PTU, and pass information about failed `service_type` to `ServiceStatusUpdateHandlerListener`

Reply:
@mked-luxoft completely unclear interfaces. We need to rename them of join.

Reply:
@LuxoftAKutsan fixed in 4a605338d184abfca2cd4da602518727e29cb9e8